### ISSUE TITLE
throwing SkipException sets iTestResult status to Failure instead of Skip

### ISSUE THREAD

**Issue by xlenz** [testng-team/testng#1632]
### Expected behavior
previously I had in beforeInvocation:

```
throw new SkipException("Skipped due to previous failure of: " + failedTestName);
```

So I have TestResult.SKIP in afterInvocation.

Discussion started here: https://github.com/cbeust/testng/pull/1611

### Actual behavior
After upgrading to 6.13.1 it stopped working, because `iTestResult.getStatus()` is now `TestResult.FAILURE` instead of `TestResult.SKIP` in afterInvocation.

to workaround this I'm setting status explicitly before throwing SkipException:
```
iTestResult.setStatus(TestResult.SKIP);
throw new SkipException("Skipping the test case");
```

### Is the issue reproductible on runner?

- [x] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

### Test case sample

```java
// listener
public class TestNGExecutionListener implements IInvokedMethodListener {
    @Override
    public void beforeInvocation(IInvokedMethod iInvokedMethod, ITestResult iTestResult) {
        throw new SkipException("skip");
    }

    @Override
    public void afterInvocation(IInvokedMethod iInvokedMethod, ITestResult iTestResult) {
        iTestResult.getStatus(); // actual Failure (2), expected Skip (3)
    }
}

// add listener
testng.addListener((ITestNGListener) new TestNGExecutionListener());
```

**Comment by McGriddle** [comment]
Setting the iTestResult.setStatus(TestResult.SKIP) before throwing the SkipException in the #beforeInvocation method causes me to see failures if the method is a BeforeXXX.

**Comment by klubi** [comment]
Is there a plan to fix this, or provide a mechanism that would allow skipping tests via configuration method?
I'd like to skip tests, not disable them with `@Test(enabled=false)`. Outcome is similar, but in terms of reporting (ex system tests) difference is huge.

**Comment by klubi** [comment]
Hi

I see this was fixed, and merged to master. Any chance this get's released anytime soon?

**Comment by krmahadevan** [comment]
ping @cbeust - Please chime in.

**Comment by klubi** [comment]
@krmahadevan Can I get a confirmation this is fixed in 7.3.0 release?
I can still see tests being marked FAILED instead of SKIP when listener throws `SkipException`.

**Comment by krmahadevan** [comment]
@klubi - Are you seeing this not working in `7.3.0` ? This version has been released.

**Comment by klubi** [comment]
I'm on `7.3.0`, and when my listeners throw SkipException tests are still marked as Failed.

When gradle logging is enabled I see something like this:

```
11:47:59  Test Suite >  Tests > com.aaa.bbbTest > shouldDoSomething FAILED
11:47:59      org.testng.SkipException
11:47:59  
11:47:59  Test Suite >  Tests > com.aaa.bbbTest > shouldDoSomething SKIPPED
```

**Comment by krmahadevan** [comment]
@klubi - Do you have a sample that you can share so that we can investigate on this ?

**Comment by klubi** [comment]
@krmahadevan let me prepare one

**Comment by klubi** [comment]
@krmahadevan 
https://github.com/klubi/vsc-testng7-sample

I re-used sample project I created for issue with VS Code.
I created `ShouldSkipTest` class, and `testSuite.xml`, and wrapped it with `runSkiptest` gradle task.

**Comment by klubi** [comment]
@krmahadevan any update on this?
It seems to still be an issue.

**Comment by sbrannen** [comment]
I believe this is effectively a duplicate of https://github.com/spring-projects/spring-framework/issues/26387.

See https://github.com/spring-projects/spring-framework/issues/26387#issuecomment-760326643 for further details.

**Comment by krmahadevan** [comment]
@klubi - I haven't gotten around to looking at this issue yet. The last few months have been extremely hectic at work front. I hope to get things sorted out at work front in a couple of weeks after which I should have some breather time. Transitioning into a full fledged backend developer at work with zero knowledge on Spring is taking a toll on my time, while still juggling the current role's responsibilities.

**Comment by juherr** [comment]
@krmahadevan No need to learn Spring. You can have a look at the test cases I added at #2455. They are based on the sample provided by @sbrannen

**Comment by krmahadevan** [comment]
@juherr - Lol.. NO that was not what I meant. I was meaning to say that since at work, I am trying to transition into a backend developer, I am having to invest a lot of time trying to figure out things in Spring etc., and hence I haven't been able to find a lot of time to get to take a look at this issue. btw thanks for adding that sample. I will take it up from there hopefully over this weekend.

**Comment by juherr** [comment]
@krmahadevan Oops, sorry I didn't catch it. BTW I've just added some new test cases. I think there are some relations with #2423 too.

**Comment by sbrannen** [comment]
@krmahadevan, thanks for pushing the fix for this!

I don't see a milestone assigned. Is this expected to be included in a 7.3.x release?

**Comment by krmahadevan** [comment]
@sbrannen - `7.3.0` is already released. This should be part of `7.4.0` (which would be the next upcoming release for TestNG)

@juherr - I think we should start leveraging the milestone capability in github so that our users exactly know as to when would a fix be available. Any thoughts on how to go about doing this ?

**Comment by krmahadevan** [comment]
@sbrannen - Thank you for bringing up this. I will start adding this milestone for all the issues that would be part of `7.4.0` going forward.

**Comment by juherr** [comment]
> I think we should start leveraging the milestone capability in github so that our users exactly know as to when would a fix be available. Any thoughts on how to go about doing this ?

@krmahadevan Good idea 👍

**Comment by krmahadevan** [comment]
@juherr - As a first step, i went ahead and added 7.4.0 as a milestone to all defects that we have fixed so far after we released 7.3.0. Going forward when we close an issue, we should ensure we update the milestone to whatever is the current upcoming version. That way this mapping would be there all the time.

**Comment by sbrannen** [comment]
Glad to you see adopting GitHub milestones.

The community thanks you! 👍
